src/configure/clients/azureDevOpsClient.ts

```diff
@@ -0,0 +1,240 @@
+import Mustache = require('mustache');
+
+import { ServiceClientCredentials, ServiceClient, UrlBasedRequestPrepareOptions } from 'ms-rest';
+
+import { WizardInputs } from '../model/models';
+
+// TO-DO: add handling failure cases
+// either throw here or analyze in the calling service layer for any errors;
+// for the second declare a model
+export class AzureDevOpsClient {
+    private serviceClient: ServiceClient;
+
+    constructor(credentials: ServiceClientCredentials) {
+        this.serviceClient = new ServiceClient(credentials);
+    }
+
+    public async getRepositoryDetails(repositoryName: string, organizationName?: string, projectName?: string): Promise<any> {
+        return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: `https://dev.azure.com/${organizationName}/${projectName}/_apis/git/repositories/${repositoryName}`,
+            headers: {
+                "Content-Type": "application/json",
+            },
+            method: "GET",
+            queryParameters: {
+                "api-version": "5.0"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async listOrganizations(): Promise<{count: number, value: [{accountId: string, accountName: string, accountUri: string, properties: {}}]}> {
+        return this.getConnectionData()
+            .then((connectionData) => {
+                return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+                    url: "https://app.vssps.visualstudio.com/_apis/accounts",
+                    headers: {
+                        "Content-Type": "application/json"
+                    },
+                    method: "GET",
+                    queryParameters: {
+                        "memberId": connectionData.authenticatedUser.id,
+                        "api-version": "5.0"
+                    },
+                    deserializationMapper: null,
+                    serializationMapper: null
+                });
+            });
+    }
+
+    public async listProjects(organizationName: string): Promise<any> {
+        return this.serviceClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: `https://${organizationName}.visualstudio.com/_apis/projects`,
+            headers: {
+                "Content-Type": "application/json"
+            },
+            method: "GET",
+            queryParameters: {
+                "includeCapabilities": "true"
+            },
+            deserializationMapper: null,
+            serializationMapper: null
+        });
+    }
+
+    public async listServiceConnections(type: string, organizationName: string, projectName: string): Promise<any> {
```

Comment:
are you using the type here?

Reply:
removing the function as it is not used not due to spec changes